new endpoint to manually add evm transaction hash & relevant address (#5593)
rotki/rotki: docs/api.rst

   :statuscode 404: There is no event with the provided identifier or the event has no details to be returned.
   :statuscode 409: No user is currently logged in.
   :statuscode 500: Internal rotki error.

Add EVM Transaction By Hash
================================

.. http:put:: /api/(version)/blockchains/evm/transactions/add-hash

>    Doing a PUT on this endpoint will add an evm transaction with the provided associated address to the database.

Reviewer: ```suggestion
   Doing a PUT on this endpoint will add an EVM transaction to the database and link it to the provided address
```

      Host: localhost:5042
      Content-Type: application/json;charset=UTF-8

      {
        "evm_chain": "ethereum",
        "tx_hash": "0x65d53653c584cde22e559cec4667a7278f75966360590b725d87055fb17552ba",
        "associated_address": "0xb8553D9ee35dd23BB96fbd679E651B929821969B",
        "async_query": true
      }

   :reqjson bool async_query: Boolean denoting whether this is an asynchronous query or not.
   :reqjson str evm_chain: The name of the evm chain for the transaction to the added e.g. ``"ethereum"``, ``"optimism"`` etc.
   :reqjson str tx_hash: The hash of the transaction to be added.
   :reqjson str associated_address: The address to be associated with the transaction. The address must be one that is already tracked by rotki.

   **Example Response**:

   .. sourcecode:: http

      HTTP/1.1 200 OK
      Content-Type: application/json

      {
          "result": true
          "message": ""
      }

   :resjson bool result: It contains a boolean representing the status of the request.

   :statuscode 200: The transaction was saved successfully.
   :statuscode 400: Provided JSON is in some way malformed.

>    :statuscode 409: No user is currently logged in. Transaction hash is already present in database

Author: but that is caught by the serializer validation

>    :statuscode 409: No user is currently logged in. Transaction hash is already present in database

Reviewer: yes but is a reason for the error to appear and is what the frontend will sea so it needs to appear in the docs too
